.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: Tag on Github and Release to RubyGems
+on: [workflow_dispatch]
+
+jobs:
+  release:
+    name: Release to RubyGems
+    runs-on: ubuntu-latest
+    # Ideally, we would require successful tests before
+    # allowing to publish. Because the tests need to be
+    # in a separate workflow file, we can't reference it
+    # here.
+    # needs: test
```

Comment:
But we could add a `run` block here to run the tests, no? Or am I misunderstanding something?

Reply:
I think this thread tackles the issue fairly well: https://github.community/t/depend-on-another-workflow/16311/3. The key point: Because each PR needs to have passing tests, you know that master will implicitly pass as well.

Reply:
Well the PR might have passing tests, but if merging occurs (in case a branch is not fully rebased to master), this does not imply that the tests also pass for master. Do you think we can safely ignore that @andyundso ?

Technically, we could add another `run` block to run the tests, but then we would be forced to basically duplicate the test workflow, and that hindered me from doing so.

Reply:
You can force this within the Github settings that every PRs branch needs to be up-to-date with master before it can be merged.

Otherwise, you can still port the pipeline to Semaphore, if deduplication is a concern.

Reply:
I added a branch protection rule to enforce up-to-date PRs before merging.